### PR TITLE
signal-cli 0.6.2 (new formula)

### PR DESCRIPTION
**Pull request by aerickson**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/signal-cli.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  desc "CLI and dbus interface for WhisperSystems/libsignal-service-java"
+  homepage "https://github.com/AsamK/signal-cli"
+  url "https://github.com/AsamK/signal-cli/releases/download/v0.6.2/signal-cli-0.6.2.tar.gz"
+  sha256 "d917507211b3419e271674df7f03c8189921e2e78a389479a612847ddc8f1614"
+  depends_on :java => "1.7+"
+
+  def install
+    # pattern taken from activemq
+    libexec.install Dir["lib", "bin"]
+    (bin/"signal-cli").write_env_script libexec/"bin/signal-cli", Language::Java.java_home_env("1.6+")
```
Comment (fxcoudert) [inline Formula/signal-cli.rb]: Is it Java 6 (above) or Java 7 (here)?

File: Formula/signal-cli.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class SignalCli < Formula
+  desc "CLI and dbus interface for WhisperSystems/libsignal-service-java"
+  homepage "https://github.com/AsamK/signal-cli"
+  url "https://github.com/AsamK/signal-cli/releases/download/v0.6.2/signal-cli-0.6.2.tar.gz"
+  sha256 "d917507211b3419e271674df7f03c8189921e2e78a389479a612847ddc8f1614"
+  depends_on :java => "1.7+"
+
+  def install
+    # pattern taken from activemq
```
Comment (fxcoudert) [inline Formula/signal-cli.rb]: Remove comment

File: Formula/signal-cli.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+class SignalCli < Formula
+  desc "CLI and dbus interface for WhisperSystems/libsignal-service-java"
+  homepage "https://github.com/AsamK/signal-cli"
+  url "https://github.com/AsamK/signal-cli/releases/download/v0.6.2/signal-cli-0.6.2.tar.gz"
+  sha256 "d917507211b3419e271674df7f03c8189921e2e78a389479a612847ddc8f1614"
+  depends_on :java => "1.7+"
```
Comment (fxcoudert) [inline Formula/signal-cli.rb]: Blank line above `depends_on`

Comment (fxcoudert) [reply Formula/signal-cli.rb]: Also please add a `bottle :unneeded` line

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/signal-cli.rb (hunk 1 above)
Comment (aerickson) [reply Formula/signal-cli.rb]: 1.6. Fixed.

File: Formula/signal-cli.rb (hunk 2 above)
Comment (aerickson) [reply Formula/signal-cli.rb]: Done.

File: Formula/signal-cli.rb (hunk 3 above)
Comment (aerickson) [reply Formula/signal-cli.rb]: Both done.

File: Formula/signal-cli.rb (hunk 1 above)
Comment (aerickson) [reply Formula/signal-cli.rb]: signal-cli can run on java 1.6 if the patchlevel is high enough, but homebrew's CI instances aren't there and it's not possible to require a specific patch level. Going to 1.7+.

File: Formula/signal-cli.rb (hunk 3 above)
Comment (javian) [reply Formula/signal-cli.rb]: There should be an empty line between bottle and depends_on too so almost there =)

File: Formula/signal-cli.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  bottle :unneeded
+  depends_on :java => "1.7+"
+
+  def install
+    libexec.install Dir["lib", "bin"]
+    (bin/"signal-cli").write_env_script libexec/"bin/signal-cli", Language::Java.java_home_env("1.7+")
+  end
+
+  test do
+    # checks class loading is working and version is correct
```
Comment (fxcoudert) [inline Formula/signal-cli.rb]: Any chance we can have a test that does a little more? Like try to connect / perform some operation, and check that it works (or fails)?

Comment (aerickson) [reply Formula/signal-cli.rb]: The `signal-cli link` command was the only one I found that exercised the cryptographic aspects and didn't require credentials configured. It was working on Homebrew CI, but something changed recently and it's now failing (network operations seem prohibited), so I removed it in 7a3900e. All of the other commands require network or credentials.

File: Formula/signal-cli.rb (hunk 3 above)
Comment (aerickson) [reply Formula/signal-cli.rb]: Added. :)

File: Formula/signal-cli.rb (hunk 4 above)
Comment (fxcoudert) [reply Formula/signal-cli.rb]: Network operations are permitted in CI.

Comment (aerickson) [reply Formula/signal-cli.rb]: OK, I'll revert that test change and we can debug networking. Thanks. :)

Comment (aerickson) [reply Formula/signal-cli.rb]: 8 seconds (vs 2) seems to have worked.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @aerickson for your contribution to Homebrew! 🎊

**Pull request merged**